front/src/applications/editor/tools/rangeEdition/tool-factory.tsx

```diff
@@ -52,6 +52,8 @@ type RangeEditionToolParams<T extends EditorRange> = {
   leftPanelComponent: ComponentType;
   canSave?: (state: RangeEditionState<T>) => boolean;
   getEventsLayers?: (context: ReadOnlyEditorContextType<RangeEditionState<T>>) => string[];
+  requiredLayers: Set<Layer>;
+  incompatibleLayers?: Layer[];
```

Comment:
Why is one a Set and not the other ?

Reply:
I used an array because I think it's more practical than a Set. I wanted to use an array for both, but it would require more change for requiredLayers, so I thought I might do it later in an other PR.

Maybe I should also ask Alexis and Benoît about why they decided to use sets instead of arrays before doing so. Is it ok for you if we merge this like this ?

Reply:
> I used an array because I think it's more practical than a Set. I wanted to use an array for both, but it would require more change for requiredLayers, so I thought I might do it later in another PR.

I figured it might be for that reason, ok :)